src/main/scala/firrtl/passes/formal/AssertSubmoduleAssumptions.scala

```diff
@@ -0,0 +1,33 @@
+
+package firrtl.passes.formal
+
+import firrtl.passes.{Pass, PullMuxes, ReplaceAccesses, ResolveFlows}
+import firrtl.ir.{Assert, Assume, Circuit, Statement}
+import firrtl.options.Dependency
+import firrtl.stage.TransformManager.TransformDependency
+import firrtl.{CircuitState, FirrtlUserException, Transform}
+
+
+object AssertSubmoduleAssumptions extends Pass {
+  override def prerequisites: Seq[TransformDependency] =
+    firrtl.stage.Forms.Deduped
+
+  override def invalidates(a: Transform): Boolean = a match {
+    case _ => false
+  }
+
```

Comment:
This should probably include the following to make sure it runs before the middle emitter. I also tend to be explicit with all dependencies as opposed to letting them inherit from `Pass` or `Transform`.

```scala
def optionalPrerequisites = Seq.empty
def optionalPrerequisiteOf = firrtl.stage.Forms.MidEmitters
```